addons/mail/static/src/model/store.js

```diff
@@ -19,6 +19,27 @@ export class Store extends Record {
         return this.recordByLocalId.get(localId);
     }
 
+    destroy() {
+        this.MAKE_UPDATE(() => {
+            const singletons = [];
+            for (const [localId, record] of this.recordByLocalId.entries()) {
+                if (this.localId !== localId && record.exists()) {
+                    if (record.Model._localId(record.Model.id, record) === undefined) {
+                        // singleton, delete after non-singletons
+                        singletons.push(record);
+                    } else {
+                        record.delete();
+                    }
+                }
+            }
+            for (const singleton of singletons) {
+                singleton.delete();
+            }
+        });
+        this.delete();
+        this._[IS_DELETED_SYM] = true;
+    }
```

Comment:
Not sure deletion of singleton at the end is still necessary. Was put before the `MAKE_UPDATE` wrap.